cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java

```diff
@@ -493,169 +299,37 @@ public void dropPartition(HttpRequest request, HttpResponder responder,
-   * @throws UnsupportedTypeException
-   */
-  public static String generateStreamCreateStatement(Id.Stream streamId, Location location, Schema bodySchema)
-    throws UnsupportedTypeException {
-    String name = streamId.getName();
-    // schema of a stream is always timestamp, headers, and then the schema of the body.
-    List<Schema.Field> fields = Lists.newArrayList(
-      Schema.Field.of("ts", Schema.of(Schema.Type.LONG)),
-      Schema.Field.of("headers", Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING))));
-    fields.addAll(bodySchema.getFields());
-    Schema schema = Schema.recordOf("streamEvent", fields);
-
-    Map<String, String> serdeProperties = ImmutableMap.of(
-      Constants.Explore.STREAM_NAME, name,
-      Constants.Explore.STREAM_NAMESPACE, streamId.getNamespaceId());
-
-    return new CreateStatementBuilder(name, getStreamTableName(streamId))
-      .setSchema(schema)
-      .setLocation(location)
-      .setTableComment("CDAP Stream")
-      .buildWithStorageHandler(Constants.Explore.STREAM_STORAGE_HANDLER_CLASS, serdeProperties);
-  }
-
-  public static String generateFileSetCreateStatement(Id.DatasetInstance datasetId, Dataset dataset,
-                                                      Map<String, String> properties) throws IllegalArgumentException {
-
-    String tableName = getHiveTableName(datasetId.getId());
-    Map<String, String> tableProperties = FileSetProperties.getTableProperties(properties);
-
-    Location baseLocation;
-    Partitioning partitioning = null;
-    if (dataset instanceof PartitionedFileSet) {
-      partitioning = ((PartitionedFileSet) dataset).getPartitioning();
-      baseLocation = ((PartitionedFileSet) dataset).getEmbeddedFileSet().getBaseLocation();
-    } else {
-      baseLocation = ((FileSet) dataset).getBaseLocation();
-    }
-
-    CreateStatementBuilder createStatementBuilder = new CreateStatementBuilder(datasetId.getId(), tableName)
-      .setLocation(baseLocation)
-      .setPartitioning(partitioning)
-      .setTableProperties(tableProperties);
-
-    String format = FileSetProperties.getExploreFormat(properties);
-    if (format != null) {
-      // for text and csv, we know what to do
-      Preconditions.checkArgument("text".equals(format) || "csv".equals(format),
-                                  "Only text and csv are supported as native formats");
-      String schema = FileSetProperties.getExploreSchema(properties);
-      Preconditions.checkNotNull(schema, "for native formats, explore schema must be given in dataset properties");
-      String delimiter = null;
-      if ("text".equals(format)) {
-        delimiter = FileSetProperties.getExploreFormatProperties(properties).get("delimiter");
-      } else if ("csv".equals(format)) {
-        delimiter = ",";
+  private Dataset instantiateDataset(Id.DatasetInstance datasetInstanceId, HttpResponder responder) throws Exception {
```

Comment:
Is there duplication between this and  ``ExploreServiceUtils.instantiateDataset()``?